src/stcal/jump/jump.py

```diff
@@ -0,0 +1,206 @@
+import time
+import logging
+import numpy as np
+
+from . import twopoint_difference as twopt
+
+import multiprocessing  # 05/18/21: TODO- commented out now; reinstate later
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+# 05/18/21 :TODO instead import these DQ flags
+DO_NOT_USE = 1
+NO_GAIN_VALUE = 524288
+
+
+def detect_jumps(frames_per_group, data, gdq, pdq, err,
+                 in_refout, gain_2d, readnoise_2d, rejection_thresh,
+                 three_grp_thresh, four_grp_thresh, max_jump_to_flag_neighbors,
+                 min_jump_to_flag_neighbors, flag_4_neighbors):
+    """
+    This is the high-level controlling routine for the jump detection process.
+    It loads and sets the various input data and parameters needed by each of
+    the individual detection methods and then calls the detection methods in
+    turn.
+
+    Note that the detection methods are currently set up on the assumption
+    that the input science and error data arrays will be in units of
+    electrons, hence this routine scales those input arrays by the detector
+    gain. The methods assume that the read noise values will be in units
+    of DN.
+
+    The gain is applied to the science data and error arrays using the
+    appropriate instrument- and detector-dependent values for each pixel of an
+    image.  Also, a 2-dimensional read noise array with appropriate values for
+    each pixel is passed to the detection methods.
+
+    Parameters
+    ----------
+    frames_per_group : int
+        number of frames per group
+
+    data : float, 4D array
+        science array
+
+    gdq : int, 4D array
+        group dq array
+
+    pdq : int, 2D array
+        pixelg dq array
+
+    err : float, 4D array
+        error array
+
+    in_refout : float, 4D array
+        reference output
+
+    gain_2d : float, 2D array
+        gain for all pixels
```

Comment:
and units here, and everywhere else where the input has a unit

Reply:
See above.